### ISSUE TITLE
Memory Allocation / Core Dump

### ISSUE THREAD

**Issue by greenstick** [gmarcais/Jellyfish#81]
Hi,

I'm attempting to run Quorum and encountering an issue with memory allocation. I looked into it and it seems Jellyfish is the source; I've replicated the error calling the following command:

```
jellyfish count -s 200000000000000 -m 24 -t 16 -o data.jf data.fastq
```

The error being output is:
```
terminate called after throwing an instance of 'jellyfish::large_hash::array_base<jellyfish::mer_dna_ns::mer_base_static<unsigned long, 0>, unsigned long, atomic::gcc, jellyfish::large_hash::array<jellyfish::mer_dna_ns::mer_base_static<unsigned long, 0>, unsigned long, atomic::gcc, allocators::mmap> >::ErrorAllocation'
  what():  Failed to allocate 200000000000000 bytes of memory
Aborted (core dumped)
```

No matter how much memory I allocate (the compute node on the cluster has 256GB at its disposal), or how I allocate it (e.g. 200G), I appear to be getting this error.

The core dump indicates that the program exited with signal 6 (No such device or address (ENXIO)):
```
Core was generated by `jellyfish count -s 200000000000000 -m 24 -t 12 -o '.
Program terminated with signal 6, Aborted.
#0  0x0000003b7ac325e5 in ?? ()
```

This is happening on a Linux cluster running CentOS 6.5:
```
Linux version 2.6.32-504.30.3.el6.x86_64 (mockbuild@c6b8.bsys.dev.centos.org) (gcc version 4.4.7 20120313 (Red Hat 4.4.7-11) (GCC) ) #1 SMP Wed Jul 15 10:13:09 UTC 2015
```

With an updated GCC compiler:
```
gcc (Homebrew gcc 5.3.0) 5.3.0
Copyright (C) 2015 Free Software Foundation, Inc.
```

Is this a known issue / is there a known workaround?

Thanks!

**Comment by greenstick** [comment]
May have got this to work with a lower value, the value for -s in the example above is much too high (though I did try several other values and received similar results.)

**Comment by ViriatoII** [comment]
Well, I've used -s 1G and it still dies with 100 G ram..

**Comment by greenstick** [comment]
Unfortunate this doesn't seem to have been resolved. Are you using a distributed file system? We were using Lustre on our cluster and, though I'm not sure the instance I have in mind relates to Jellyfish, I recall we had an issue with how some C code was opening/closing files. I think it may have been that a lot of file handles were being generated and it was gumming the distributed file systems' I/O. 

It may also be that the space complexity of one of Jellyfish's routines is very high – I also remember for some of the software being used in our analysis we had to move to 512GB nodes. Apologies I can't provide more details - this was over 2 years ago. Good luck!
